From 10d816b5e5a76bc29660991eaf7f751bba4a7cd8 Mon Sep 17 00:00:00 2001
From: Meron Cohen <mecohen@habana.ai>
Date: Tue, 11 Jun 2024 11:45:53 +0300
Subject: [PATCH] [FSW-4709] scripts: update fs1 post-build scripts

1. The regexp used in hl_create_zephyr_bins_config relies on the macros
defined in fs1_mem_addr_dt.h. In FS1, contrary to Gaudi3, both
ARC_MGMT_IO_ICCM_ and ARC_MGMT_ICCM_ macros are defined. This causes
value duplication strings assigned to off and sz variables. For example:
sz = "0x100 0x100" instead of "0x100".
2. The arcpid implementation isn't ready yet, so there was a need to
comment out temporarily more lines related to it for the script to pass.

Change-Id: Ib4a84cd26da88902961b51305cd34a0de2b7756a
Signed-off-by: Meron Cohen <mecohen@habana.ai>
---

diff --git a/board/habanalabs/fs1_arc/fs1-image-utility.sh b/board/habanalabs/fs1_arc/fs1-image-utility.sh
index 9b357d8..4888079 100755
--- a/board/habanalabs/fs1_arc/fs1-image-utility.sh
+++ b/board/habanalabs/fs1_arc/fs1-image-utility.sh
@@ -152,8 +152,8 @@
 		bins_path=$BINS_CONFIG_DIR/${HL_TARGET}_${module}.bins
 		echo "name,address,size_kb,align_bytes" > $bins_path
 		while read -r section; do
-			off=`grep ${module^^}.*$section.*_OFFSET $mem_sections_hdr_path | sed 's/[[:space:]]\+/ /g' | cut -d ' ' -f3`
-			sz=`grep ${module^^}.*$section.*_SIZE $mem_sections_hdr_path | sed 's/[[:space:]]\+/ /g' | cut -d ' ' -f3`
+			off=`grep ${module^^}_$section.*_OFFSET $mem_sections_hdr_path | sed 's/[[:space:]]\+/ /g' | cut -d ' ' -f3`
+			sz=`grep ${module^^}_$section.*_SIZE $mem_sections_hdr_path | sed 's/[[:space:]]\+/ /g' | cut -d ' ' -f3`
 			echo ${section,,},0x$off,$sz,16 >> $bins_path
 		done <<< `grep "_SECTION" $mem_sections_hdr_path | grep ${module^^} | cut -d '_' -f3 | sort -u`
 	done <<< `grep "_SECTION" $mem_sections_hdr_path | cut -d '_' -f2 | sort -u | tr [:upper:] [:lower:]`
diff --git a/board/habanalabs/fs1_arc/post-image-fit.sh b/board/habanalabs/fs1_arc/post-image-fit.sh
index 6abc0b5..42a3085 100755
--- a/board/habanalabs/fs1_arc/post-image-fit.sh
+++ b/board/habanalabs/fs1_arc/post-image-fit.sh
@@ -19,8 +19,9 @@
 	local FILEPATH=$1
 
 	# copy binaries from other HL targets required for boot fit creation
-	cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-pid-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-pid-iccm.bin.le
-	cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-pid-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-pid-dccm.bin.le
+	# TEMPORARY: arcpid implementation isn't ready yet, so comment it out.
+	#cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-pid-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-pid-iccm.bin.le
+	#cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-pid-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-pid-dccm.bin.le
 	cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-mgmt-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-iccm.bin.le
 	cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-mgmt-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-dccm.bin.le
 
@@ -33,21 +34,25 @@
 popd
 
 MGMT_STAT=${STATS_DIR}/zephyr_mgmt.stat
-PID_STAT=${STATS_DIR}/zephyr_pid.stat
+# TEMPORARY: arcpid implementation isn't ready yet, so comment it out.
+#PID_STAT=${STATS_DIR}/zephyr_pid.stat
 
 # Extract the start address from stat file
 ENTRY_ADDRESS=$(cat $MGMT_STAT | grep "Entry point address" | head -1)
 MGMT_STARTADDR=$(echo $ENTRY_ADDRESS | awk -F: '{ print $2 }' | xargs)
-ENTRY_ADDRESS=$(cat $PID_STAT | grep "Entry point address" | head -1)
-PID_STARTADDR=$(echo $ENTRY_ADDRESS | awk -F: '{ print $2 }' | xargs)
+# TEMPORARY: arcpid implementation isn't ready yet, so comment it out.
+#ENTRY_ADDRESS=$(cat $PID_STAT | grep "Entry point address" | head -1)
+#PID_STARTADDR=$(echo $ENTRY_ADDRESS | awk -F: '{ print $2 }' | xargs)
 
 # Start address is only 2 bytes. Perform OR operation with
 # ARC_PID_ICCM_LO to get 32 bit address and XX_STARTADDR_HI
 # is same as ARC_XX_ICCM_HI.
-PID_STARTADDR_LO=$((ARC_PID_ICCM_LO | PID_STARTADDR))
+# TEMPORARY: arcpid implementation isn't ready yet, so comment it out.
+#PID_STARTADDR_LO=$((ARC_PID_ICCM_LO | PID_STARTADDR))
 MGMT_STARTADDR_LO=$((ARC_MGMT_ICCM_LO | MGMT_STARTADDR))
 
-ITS_SED_PARAMS+="s/@PID_STARTADDR_HI@/${ARC_PID_ICCM_HI}/g;"
-ITS_SED_PARAMS+="s/@PID_STARTADDR_LO@/${PID_STARTADDR_LO}/g;"
+# TEMPORARY: arcpid implementation isn't ready yet, so comment it out.
+#ITS_SED_PARAMS+="s/@PID_STARTADDR_HI@/${ARC_PID_ICCM_HI}/g;"
+#ITS_SED_PARAMS+="s/@PID_STARTADDR_LO@/${PID_STARTADDR_LO}/g;"
 ITS_SED_PARAMS+="s/@MGMT_STARTADDR_HI@/${ARC_MGMT_ICCM_HI}/g;"
 ITS_SED_PARAMS+="s/@MGMT_STARTADDR_LO@/${MGMT_STARTADDR_LO}/g;"
